contracts/base/ModuleManager.sol

```diff
@@ -246,6 +309,113 @@ contract ModuleManager is Storage, Receiver, IModuleManagerEventsAndErrors {
+    /// @dev Install multiple modules at once
+    /// @dev It will call module.onInstall for every initialization so it ensure the flow
+    /// consistent with the flow of the SA's that do not implement _multiTypeInstall 
+    /// and thus will call the multityped module several times
+    /// The multityped modules can not expect all the 7579 SA's to implement _multiTypeInstall and
+    /// thus should account for the flow when they are going to be called with onUnistall
+    /// for the initialization as every of the module types they declare they are 
+    /// @param module address of the module
+    /// @param initData initialization data for the module
+    function _multiTypeInstall(
+        address module,
+        bytes calldata initData
+    )
+        internal virtual
+    {
+        (uint256[] calldata types, bytes[] calldata initDatas) = initData.parseMultiTypeInitData();
+        
+        uint256 length = types.length;
+        if (initDatas.length != length) revert InvalidInput();
+
+        // iterate over all module types and install the module as a type accordingly
+        for (uint256 i; i < length; i++) {
+            uint256 theType = types[i];
+
+            /*´:°•.°+.*•´.*:˚.°*.˚•´.°:°•.°•.*•´.*:˚.°*.˚•´.°:°•.°+.*•´.*:*/
+            /*                      INSTALL VALIDATORS                    */
+            /*.•°:°.´+˚.*°.˚:*.´•*.+°.•°:´*.´•*.•°.•°:°.´:•˚°.*°.˚:*.´+°.•*/
+            if (theType == MODULE_TYPE_VALIDATOR) {
+                _installValidator(module, initDatas[i]);
+            }
+            /*´:°•.°+.*•´.*:˚.°*.˚•´.°:°•.°•.*•´.*:˚.°*.˚•´.°:°•.°+.*•´.*:*/
+            /*                       INSTALL EXECUTORS                    */
+            /*.•°:°.´+˚.*°.˚:*.´•*.+°.•°:´*.´•*.•°.•°:°.´:•˚°.*°.˚:*.´+°.•*/
+            else if (theType == MODULE_TYPE_EXECUTOR) {
+                _installExecutor(module, initDatas[i]);
+            }
+            /*´:°•.°+.*•´.*:˚.°*.˚•´.°:°•.°•.*•´.*:˚.°*.˚•´.°:°•.°+.*•´.*:*/
+            /*                       INSTALL FALLBACK                     */
+            /*.•°:°.´+˚.*°.˚:*.´•*.+°.•°:´*.´•*.•°.•°:°.´:•˚°.*°.˚:*.´+°.•*/
+            else if (theType == MODULE_TYPE_FALLBACK) {
+                _installFallbackHandler(module, initDatas[i]);
+            }
+            /*´:°•.°+.*•´.*:˚.°*.˚•´.°:°•.°•.*•´.*:˚.°*.˚•´.°:°•.°+.*•´.*:*/
+            /*          INSTALL HOOK (global or sig specific)             */
+            /*.•°:°.´+˚.*°.˚:*.´•*.+°.•°:´*.´•*.•°.•°:°.´:•˚°.*°.˚:*.´+°.•*/
+            else if (theType == MODULE_TYPE_HOOK) {
+                _installHook(module, initDatas[i]);
+            }
+        }
+    }
+
+        
+    /// @notice Checks if an enable mode signature is valid.
+    /// @param digest signed digest.
+    /// @param sig Signature.
+    function _checkEnableModeSignature(bytes32 digest, bytes calldata sig) internal view {
+        address enableModeSigValidator = address(bytes20(sig[0:20]));
```

Comment:
this could be any of the installed validators?

Reply:
yes